src/Microsoft.Kusto.ServiceLayer/QueryExecution/Batch.cs

```diff
@@ -369,6 +367,16 @@ private async Task ExecuteOnce(ReliableDataSourceConnection conn, CancellationTo
 
                 } while (reader.NextResult());
 
+                if (resultSets.Count > 3)
+                {
+                    resultSets.RemoveRange(resultSets.Count - 3, 3);
+                }
```

Comment:
QueryExecution is not the right place to do this. For the Kusto.ServiceLayer we are localizing all Kusto specific changes to the DataSource folder. Kusto specific logic should not be sprinkled across the rest of the code which we've kept generic.

A great place to add this is in the KustoResultsReader.cs.

Reply:
Done